PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -2226,6 +2237,62 @@ class PurchasesTests: XCTestCase {
         expect(self.storeKitWrapper.finishCalled).toEventually(beFalse())
     }
 
+    func testDoesntPostTransactionsIfAutoSyncPurchasesSettingIsOffInObserverMode() throws {
+        systemInfo = try MockSystemInfo(platformFlavor: nil,
+                                        platformFlavorVersion: nil,
+                                        finishTransactions: false,
+                                        dangerousSettings: DangerousSettings(autoSyncPurchases: false))
+        initializePurchasesInstance(appUserId: nil)
+
+        let product = StoreProduct(sk1Product: MockSK1Product(mockProductIdentifier: "com.product.id1"))
+        self.purchases?.purchase(product: product) { (_, _, _, _) in
```

Comment:
We've been trying to break this pattern and not use `?` in this scenario, as it would silently not do anything. You can remove it since it's implicitly unwrapped.